src/App.jsx

```diff
@@ -0,0 +1,285 @@
+/* eslint-disable array-callback-return */
+
+import "./App.css";
+import { v4 as uuidv4 } from "uuid";
+import React, { useState, useEffect } from "react";
+import Todos from "./components/Todos";
+import Header from "./components/Header";
+import Icon from "./components/Icon";
+import Tag from "./components/Tag";
+import Toaster from "./components/Toaster";
+import { supabase } from "./config/apiClient";
+import classNames from "classnames";
+
+export const AppContext = React.createContext();
+
+function App() {
+  const [todos, setTodos] = useState([]);
+
+  const [toasts, setToasts] = useState([]);
+  const [showEmpty, setShowEmpty] = useState(true);
+  const [showBigSpinner, setShowBigSpinner] = useState(false);
+  const [searchShow, setSearchShow] = useState(false);
+  const [flag, setFlag] = useState("all");
+  const [search, setSearch] = useState("");
+  const [dataCount, setDataCount] = useState(0);
+  const [splash, setSplash] = useState(true);
+  const [flagChange, setFlagChange] = useState(false);
+  const [prevFlag, setPrevFlag] = useState("all");
+  const [progress, setProgress] = useState(true);
+  //flag
+  const flagHandler = (e) => {
+    if (flag !== e) {
+      setFlagChange(true);
+      setPrevFlag(flag);
+      setFlag(e);
+    }
+  };
+  const removeCompleteFromIncomplete = (id) => {
+    const filtertodos = todos.filter((todo) => todo.id !== id);
+    setTodos(filtertodos);
+  };
+  //search value
+  const searchvalue = (e) => {
+    if (e.length > 2) {
+      setShowBigSpinner(true);
+      setSearch(e);
+
+      setShowBigSpinner(false);
+    }
+  };
+
+  //search toggle
+  const SearchToggle = () => {
+    if (search !== "") {
+      setShowBigSpinner(true);
+      setSearch("");
+      setShowBigSpinner(false);
+    }
+    setSearchShow(!searchShow);
+  };
+
+  // useEffect
+  useEffect(() => {
+    const fetchData = async (e) => {
+      setShowBigSpinner(true);
+      let newToast;
+      if (flag === "all") {
+        const { data, error } = await supabase
+          .from("ReactTodo")
+          .select()
+          .ilike("name", `%${search}%`)
+          .order("id", { ascending: false });
+
+        if (error) {
+          if (flagChange) {
+            setFlag(prevFlag);
+          }
+          if (splash) {
+            setProgress(false);
+            setSplash(false);
+          }
+        } else {
+          setTodos([]);
+          setTodos(data);
+          if (splash) {
+            setDataCount(data.length);
+            setProgress(false);
+            setTimeout(() => {
+              setSplash(false);
+            }, 500);
+          }
+        }
+        newToast = {
+          id: uuidv4(),
+          type: error ? "error" : "success",
+          message: error ? error.message : "All Data fetched",
```

Comment:
maybe we can keep the message dynamically in a variable and just set that here. If error, we set the message to error.message otherwise a success message